tapchannel/aux_traffic_shaper.go

```diff
@@ -297,6 +319,31 @@ func (s *AuxTrafficShaper) PaymentBandwidth(htlcBlob,
 	return availableBalanceMsat, nil
 }
 
+// ComputeLocalBalance combines the given commitment state with the HtlcView to
+// produce the available local balance with accuracy.
+func ComputeLocalBalance(commitment *cmsg.Commitment,
+	htlcView lnwallet.AuxHtlcView) (uint64, *DecodedView, error) {
+
+	// Let's get the current local and remote asset balances of the channel
+	// as reported by the latest commitment.
+	localBalance := cmsg.OutputSum(commitment.LocalOutputs())
+	remoteBalance := cmsg.OutputSum(commitment.RemoteOutputs())
+
+	// With the help of the latest HtlcView, let's calculate a more precise
+	// local balance. This is useful in order to not forward HTLCs that may
+	// never be settled. Other HTLCs that may also call into this method are
+	// not yet registered to the commitment, so we need to account for them
+	// manually.
+	computedLocal, _, decodedView, _, err := ComputeView(
```

Comment:
This is still missing the explicit setting of `htlcView.NextHeight = 0` and the explanation around that.